File: CS/HatchAppointmenStatus.cs
using System.Drawing;
using System.Drawing.Drawing2D;
using DevExpress.XtraScheduler;
using DevExpress.XtraScheduler.Native;

namespace ScheduleriCalendarExchangeCustomization {
    public class HatchAppointmenStatus : AppointmentStatus {
        private Color foreColor;
        private Color backColor;

        public HatchAppointmenStatus(int id, Color foreColor, Color backColor, string caption)
            : base((AppointmentStatusType)id, caption) {

            this.foreColor = foreColor;
            this.backColor = backColor;
        }

        protected override void Visit(IUserInterfaceObjectVisitor visitor) {
            IBrushAccessor brushAccessor = (IBrushAccessor)visitor;
            brushAccessor.Brush = new HatchBrush(HatchStyle.Cross, foreColor, backColor);
        }
    }
}

File: VB/HatchAppointmenStatus.vb
Imports Microsoft.VisualBasic
Imports System.Drawing
Imports System.Drawing.Drawing2D
Imports DevExpress.XtraScheduler

Namespace ScheduleriCalendarExchangeCustomization
	Public Class HatchAppointmenStatus
		Inherits AppointmentStatus
		Private foreColor As Color
		Private backColor As Color

		Public Sub New(ByVal id As Integer, ByVal foreColor As Color, ByVal backColor As Color, ByVal caption As String)
			MyBase.New(CType(id, AppointmentStatusType), caption)

			Me.foreColor = foreColor
			Me.backColor = backColor
			MyBase.UpdateBrush()
		End Sub

		Protected Overrides Function CreateBrush() As Brush
			Return New HatchBrush(HatchStyle.Cross, foreColor, backColor)
		End Function
	End Class
End Namespace
